test/parallel/test-assert.js

```diff
@@ -409,6 +409,14 @@ assert.throws(() => {
 }, /Got unwanted exception: user message/,
               'a.doesNotThrow ignores user message');
 
+assert.throws(() => {
```

Comment:
IMHO the check should be using `try / catch` otherwise we get into a weird situation where we use the function to test itself and Alex Turing will get mad, and say 🛑